File: Controllers/BloqueController.php
<?php

    class BloqueController{
        private $model;

        //Función para conectar con el modelo que utiliza la bd
        public function __construct(){
            require_once __DIR__ . '/../Models/Bloque.php';
            $this->model = new Bloque;
        }

        //Función para guardar cuando se crea un nuevo usuario
        public function save($usuario, $contrasena){
            $id = $this->model->createUser($usuario, $contrasena);
            return ($id!=false) ? header("Location:show.php?id=" .$id) : header("Location:create.php");
        } 

        public function saveEstado($id_usuario, $estado) 
        {
                $id_usuario = (int)$id_usuario;
                $resultado = $this->model->createBlo($id_usuario, $estado);
                return $resultado === true;
        }
         
         //Función para iniciar sesión (en BloqueController.php)
        public function login($usuario, $contrasena)
        {
            $user = $this->model->login($usuario, $contrasena);
            if ($user) {
                session_start();

                $_SESSION['usuario'] = $user['usuario'];
                $_SESSION['id_usuario'] = $user['id']; 

                header("Location: button.php");

            } else {
                header("Location: login.php");
            }
        }
    }
?>

File: controllers/bloque_controller.py
from models.bloque import Bloque
from flask import session, redirect, url_for

class BloqueController:
    def __init__(self):
        self.model = Bloque()
    
    def save(self, usuario, contrasena):
        id = self.model.create_user(usuario, contrasena)
        return id
    
    def save_estado(self, id_usuario, estado):
        id_usuario = int(id_usuario)
        resultado = self.model.create_blo(id_usuario, estado)
        return resultado
    
    def login(self, usuario, contrasena):
        user = self.model.login(usuario, contrasena)
        if user:
            session['usuario'] = user['usuario']
            session['id_usuario'] = user['id']
            return True
        return False
